api/app/auto_spatial_advisory/process_fuel_type_area.py

```diff
@@ -0,0 +1,225 @@
+    """
+    Process masked fuel type layer with each advisory shape (eg fire zone unit).
+
+    :param temp_dir: A temporary location for storing intermediate files.
+    :param source_path: Path to the masked fuel type layer.
+    :param threshold: The current threshold being processed, 1 = 4k-10k, 2 = > 10k.
+    :param run_parameters_id: The RunParameter object id associated with the run_type, for_date and run_datetime of interest.
+    :param fuel_types: A list of fuel types used in the sfms system.
+    """
+    stmt = text("SELECT id, source_identifier FROM advisory_shapes;")
+    result = await session.execute(stmt)
+    rows = result.all()
+    for row in rows:
+        shape_fuel_type_path = intersect_raster_by_advisory_shape(threshold, row[0], row[1], source_path, temp_dir)
+        fuel_type_areas = calculate_fuel_type_areas(shape_fuel_type_path, fuel_types)
+        await store_advisory_fuel_stats(session, fuel_type_areas, threshold, run_parameters_id, row[0])
+
+
+async def store_advisory_fuel_stats(session: AsyncSession, fuel_type_areas: dict, threshold: int, run_parameters_id: int, advisory_shape_id: int):
+    """
+    Creates AdvisoryFuelStats objects and save them in the wps database.
+
+    :param : A dictionary keyed by fuel type code with value representing an area in square meters.
+    :param threshold: The current threshold being processed, 1 = 4k-10k, 2 = > 10k.
+    :param run_parameters_id: The RunParameter object id associated with the run_type, for_date and run_datetime of interest.
+    :param advisory_shape_id: The id of advisory shape (eg. fire zone unit) the fuel type area has been calcualted for.
+    """
+    advisory_fuel_stats = []
+    for key in fuel_type_areas:
+        advisory_fuel_stats.append(
+            AdvisoryFuelStats(advisory_shape_id=advisory_shape_id, threshold=threshold, run_parameters=run_parameters_id, fuel_type=key, area=fuel_type_areas[key])
+        )
+    await save_advisory_fuel_stats(session, advisory_fuel_stats)
+
+
+def calculate_fuel_type_areas(source_path: str, fuel_types: list[SFMSFuelType]):
+    """
+    Calculates the ground area of the raster layer at the source_path covered by each fuel type.
+
+    :param source_path: The path to a fuel type layer that has been masked based on hfi value and clipped to an advisory shape.
+    :param fuel_types: A list of fuel types from the sfms system that may be present in the source_path fuel type layer.
+    """
+    source = gdal.Open(source_path, gdal.GA_ReadOnly)
+    source_band = source.GetRasterBand(1)
+    histogram = source_band.GetHistogram()
+    combustible_fuel_type_ids = [fuel_type.fuel_type_id for fuel_type in fuel_types if fuel_type.fuel_type_id < 99 and fuel_type.fuel_type_id > 0]
+    fuel_type_areas = {}
+    for fuel_type_id in combustible_fuel_type_ids:
+        count = histogram[fuel_type_id]
+        area = count * 2000 * 2000
```

Comment:
Are the `2000` literals kilometres? Wonder if we can describe them with a variable.

Reply:
Good idea...added `FUEL_TYPE_RASTER_RESOLUTION_IN_METRES = 2000`